tasks.c

```diff
@@ -7225,13 +7225,24 @@ static void prvResetNextTaskUnblockTime( void )
                     if( uxConsumedBufferLength < ( uxBufferLength - 1 ) )
                     {
                         /* Write the rest of the string. */
-                        iSnprintfReturnValue = snprintf( pcWriteBuffer,
-                                                         uxBufferLength - uxConsumedBufferLength,
-                                                         "\t%c\t%u\t%u\t%u\r\n",
-                                                         cStatus,
-                                                         ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority,
-                                                         ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark,
-                                                         ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber ); /*lint !e586 sprintf() allowed as this is compiled with many compilers and this is a utility function only - not part of the core kernel implementation. */
+                        #if ( ( configUSE_CORE_AFFINITY == 1 ) && ( configNUMBER_OF_CORES > 1 ) )
+                            iSnprintfReturnValue = snprintf( pcWriteBuffer,
+                                                             uxBufferLength - uxConsumedBufferLength,
+                                                             "\t%c\t%u\t%u\t%u\t%x\r\n",
+                                                             cStatus,
+                                                             ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority,
+                                                             ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark,
+                                                             ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber,
+                                                             ( unsigned int ) pxTaskStatusArray[ x ].uxCoreAffinityMask ); /*lint !e586 sprintf() allowed as this is compiled with many compilers and this is a utility function only - not part of the core kernel implementation. */
```

Comment:
If we are casting this to `(unsigned int)` then in the string, should it not be `%u` instead of `%x`?

Reply:
Hi @AniruddhaKanhere I purposely made it `%x` instead of `%u` for readability/display reasons. `uxCoreAffinityMask` stores a bit mask, thus it would be more useful for users to read its value in hex.

Note: I've added a leading `0x` in the string format to make it clear it's a hex value.